elephant/statistics.py

```diff
@@ -250,6 +250,67 @@ def lv(v):
     return 3. * np.mean(np.power(np.diff(v) / (v[:-1] + v[1:]), 2))
 
 
+def cv2(v):
+    """
+    Calculate the measure of CV2 for a sequence of time intervals between 
+    events.
+
+    Given a vector v containing a sequence of intervals, the CV2 is
+    defined as:
+
+    .math $$ CV2 := \\frac{1}{N}\\sum_{i=1}^{N-1}
+
+                   \\frac{2|isi_{i+1}-isi_i|}
+                          {|isi_{i+1}+isi_i|} $$
+
+    The CV2 is typically computed as a substitute for the classical
+    coefficient of variation (CV) for sequences of events which include
+    some (relatively slow) rate fluctuation.  As with the CV, CV2=1 for
+    a sequence of intervals generated by a Poisson process.
+
+    Parameters
+    ----------
+
+    v : quantity array, numpy array or list
+        Vector of consecutive time intervals
+
+    Returns
+    -------
+    cv2 : float
+       The CV2 of the inter-spike interval of the input sequence.
+
+    Raises
+    ------
+    AttributeError :
+       If an empty list is specified, or if the sequence has less
+       than two entries, an AttributeError will be raised.
+    AttributeError :
+        Only vector inputs are supported.  If a matrix is passed to the
+        function an AttributeError will be raised.
+
+    References
+    ----------
+    ..[1] Holt, G. R., Softky, W. R., Koch, C., & Douglas, R. J. (1996). 
+    Comparison of discharge variability in vitro and in vivo in cat visual 
+    cortex neurons. Journal of neurophysiology, 75(5), 1806-1814.
+    """
+    # convert to array, cast to float
+    v = np.asarray(v)
+
+    # ensure the input ia a vector
+    if len(v.shape) > 1:
+        raise AttributeError("Input shape is larger than 1. Please provide "
+                             "a vector in input.")
+
+    # ensure we have enough entries
+    if v.size < 2:
```

Comment:
For this function, and for LV, one may ask whether it's better to return np.NaN instead of throwing an error. An error will be fool-proof, but returning a NaN (or None) will often be extremely practical.

Reply:
I agree that it is a possibility to consider. For the moment I choose to return  an error for sake of consistency with other functions in elephant (e.g. statistics.lv()). Probably what to return in case of invalid input needs to be discussed in another more general pool request involving multiple function.

Reply:
Fair enough, then this will be discussed in #115